cmake/GitSubmodule.cmake

```diff
@@ -0,0 +1,36 @@
+
+# Call to ensure that the git submodule in location `path` is loaded.
+# If the submodule is not loaded, the function will attempt to check it
+# out using git.
+# Sets the following variable `name_AVAIL` to `ON` if the submodule is
+# available on completion, or `OFF` otherwise.
+
+function(git_submodule name path)
```

Comment:
At least if we do this, there should be a flag to turn it off.

Reply:
Or at least make it only run when the user requests a target that depends on the submodule, which is currently how it works for python, but not for docs or ubenches.